src/components/view/Navigation/index.js

```diff
@@ -34,47 +30,58 @@ class Navigation extends React.Component {
 
     constructor(props) {
         super(props);
-        this.navigateToRoute = this.navigateToRoute.bind(this);
+        this.handleLogout = this.handleLogout.bind(this);
+        this.handleToggle = this.handleToggle.bind(this);
         this.themePalette = this.props.muiTheme.palette;
+        this.state = { drawerOpen: false };
+    }
+
+    handleToggle () {
+        // doesn't show nevigation unless logged in.
+        const { user } = this.props;
+        if (user.uid && user.uid.length > 0) {
+            this.setState({drawerOpen: !this.state.drawerOpen});
+        }
+    }
+
+    getMenuItemHandler (routeName) {
+        return function () {
+            this.setState({drawerOpen: false});
+            this.props.history.push(routeName);
+        }.bind(this);
     }
-    
-    buildIconMenu (actions) {
-        const { logout } = actions;
-        return (
-            <IconMenu
-                iconButtonElement={<IconButton><ExpandMoreIcon /></IconButton>}
-                anchorOrigin={{ horizontal: 'left', vertical: 'top' }}
-                targetOrigin={{ horizontal: 'left', vertical: 'top' }}
-            >
-                <MenuItem onTouchTap={logout} primaryText="Logout" />
-            </IconMenu>
-        );
-    };
 
-    navigateToRoute (someArg, evt, tabEl) {
-        // close any open event item
-        // this.toggleEventDetail();
-        // navigate
-        this.props.history.push(tabEl.props['data-route']);
-    };
+    handleLogout () {
+        this.setState({drawerOpen: false});
+        this.props.logout();
+    }
 
     render () {
-        const { user, logout } = this.props;
-        
-        const iconMenu = this.buildIconMenu({ logout });
+        const { user } = this.props;
+
+        const avatarSize = 60,
+            paddingSize = 15,
+            drawerWidth = Math.min(450, window.outerWidth * .8);
 
         return (
             <div>
-                
-                {<AppBar
-                    title={'Welcome, ' + user.displayName}
-                    iconElementRight={iconMenu}
-                >
-                </AppBar>}
-
+                <AppBar onLeftIconButtonTouchTap={this.handleToggle} title={'Welcome, ' + user.displayName}>
+                </AppBar>
+                <Drawer docked={false} width={drawerWidth} open={this.state.drawerOpen} onRequestChange={() => this.setState({drawerOpen : false})}>
+                    <div style={{backgroundColor: this.props.muiTheme.appBar.color, padding: paddingSize }}>
+                        <Avatar size={avatarSize} icon={<PersonOutlineIcon/>}/>
+                        <div style={{ color: this.props.muiTheme.appBar.textColor, paddingTop: paddingSize, paddingBottom: paddingSize }}>
+                            {'Welcome, ' + user.displayName}
```

Comment:
actually I do have one suggestion. Could we create a variable for the welcome text and use it on this line and line 68. I noticed that the screen will say `Welcome, undefined` until the user name is not undefined. So I think it would be best to create the variable above like so, and then use it here. That way it will display nothing until the name is valid

I'm also open to it just saying `Welcome` until the name is defined. Either way would be an improvement

```javascript
const welcomeText = user.displayName ? `Welcome, ${user.displayName}` : '';
```

Reply:
Actually the initial commit had code like that.  However, I didn't go with that because:

(1) The drawer, and hence the welcome message inside the drawer, are only displayed when the user has logged in.

(2) According to Evan (and the UI mock-up), the welcome message in the App Bar is just a place holder and will be replaced anyway.